src/poetry/utils/shell.py

```diff
@@ -67,8 +68,19 @@ def get(cls) -> "Shell":
         return cls._shell
 
     def activate(self, env: "VirtualEnv") -> Optional[int]:
+        activate_script = self._get_activate_script()
+        bin_dir = "Scripts" if WINDOWS else "bin"
+        activate_path = env.path / bin_dir / activate_script
+
         if WINDOWS:
-            return env.execute(self.path)
+            if self._name in ("powershell", "pwsh"):
+                args = ["-NoExit", "-File", str(activate_path)]
+            else:
+                # /K will execute the bat file and
+                # keep the cmd process from terminating
+                args = ["/K", str(activate_path)]
+            completed_proc = subprocess.run([self.path] + args)
```

Comment:
```suggestion
            completed_proc = subprocess.run([self.path, *args])
```

A minor nitpick but I think this syntax makes the intent more obvious.

Reply:
Done